types/react-router/test/examples-from-react-router-website/Auth.tsx

```diff
@@ -43,7 +43,7 @@ const fakeAuth = {
   }
 };
 
-const AuthButton = withRouter(({ history }) => (
+const AuthComponent = ({ history }: RouteComponentProps) => (
```

Comment:
If this no longer infers the type then this is a breaking change.

Reply:
That's a fair point. The inference is optional now. I'll investigate.

Reply:
Just pushed up an adjustment. By using an override to allow the partial this code was able to work as originally expected in master (and it has been reverted).